src/components/MyTemplate/index.tsx

```diff
@@ -0,0 +1,58 @@
+/* eslint-disable require-jsdoc */
+import React from "react";
+import EmptyTemplate from "./EmptyTemplate";
+import MyTemplateContainer from "./MyTemplateContainer";
+import { Flex } from "@chakra-ui/react";
+import { useStoreSelector } from "@/store/hooks";
+import { useGetTemplatesQuery } from "@/redux/api/templateApi";
+import { userData } from "@/store/reducer/userSlice";
+import { skipToken } from "@reduxjs/toolkit/dist/query";
+import { IMyTemplates } from "@/interfaces/template";
+
+function MyTemplate() {
+  const isMyTemplateOpen = useStoreSelector((state) => state.buttonToggle.isMyTemplateOpen);
+  const isCreateTemplateOpen = useStoreSelector((state) => state.buttonToggle.isCreateTemplateOpen);
+  const isCartOpen = useStoreSelector((state) => state.buttonToggle.isCartOpen);
+  const currentUserId = useStoreSelector(userData).userId;
+  const { data } = useGetTemplatesQuery(currentUserId ? currentUserId : skipToken);
+  const quantity = data?.length;
+
+  const myTemplates: IMyTemplates[] | undefined = data?.map((item: any) => {
+    return {
+      _id: item._id,
+      courtName: item.design.designName,
+      user_id: item.user_id,
+      description: item.description,
+      image: item.image,
+      status: item.status,
+      createdAt: item.createdAt,
+      tags: item.tags,
+    };
+  });
+
+  return (
+    <>
+      {isMyTemplateOpen && currentUserId && !isCreateTemplateOpen && !isCartOpen && (
```

Comment:
I should consider, because  the difference of the index.
people can click the shopping cart button when they are in the my Template page.   if I  don't consider the shopping cart, this page won't close automatically and people cannot jumpt to the shopping cart page when they click the button.

Reply:
> I think don' consider the condition of cart page

I should consider, because the difference of the index.
people can click the shopping cart button when they are in the my Template page. if I don't consider the shopping cart, this page won't close automatically and people cannot jumpt to the shopping cart page when they click the button.

Reply:
You can do the layer control in the button toggle reducer.
When you set the myTemplate state as true, you need to set cart page, share template page as false. In the other hand, when cart page or share template page are set as true, you need to close my template page.